tutorials/llm/llama-3/pruning-distillation/introduction.ipynb

```diff
@@ -0,0 +1,190 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7129d44e-0536-4e62-bdbc-0f1ad44dc84a",
+   "metadata": {},
+   "source": [
+    "2. **Set the Hugging Face Access Token**: You can obtain this from your [Hugging Face account](https://huggingface.co/docs/hub/en/security-tokens). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "481417ed-1456-4962-8f67-4350bde1aabd",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from huggingface_hub import login\n",
+    "login(token=\"<YOUR_HF_ACCESS_TOKEN>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "245eda8d-c999-431e-9ebc-5c92c4f21f3b",
+   "metadata": {},
+   "source": [
+    "3. **Obtain the dataset**: Generate the `wikitext-{train/val/test}.jsonl` splits after loading the [WikiText-103-v1](https://huggingface.co/datasets/Salesforce/wikitext/viewer/wikitext-103-v1) dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eaef2c7d-41f7-41ad-a76a-2d714e9c35de",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Split into train, test and val files\n",
+    "\n",
+    "import json\n",
+    "import os\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "# Load the WikiText-103 dataset\n",
+    "dataset = load_dataset(\"wikitext\", \"wikitext-103-v1\")\n",
+    "\n",
+    "# Define the destination folder\n",
+    "data_folder = 'wikitext-data'\n",
+    "os.makedirs(data_folder, exist_ok=True)\n",
+    "\n",
+    "# Define file paths and destination paths\n",
+    "file_paths = {\n",
+    "    'train': os.path.join(data_folder, 'wikitext-train.jsonl'),\n",
+    "    'validation': os.path.join(data_folder, 'wikitext-val.jsonl'),\n",
+    "    'test': os.path.join(data_folder, 'wikitext-test.jsonl')\n",
+    "}\n",
+    "\n",
+    "# Function to save dataset split to a JSONL file\n",
+    "def save_to_jsonl(file_path, data):\n",
+    "    with open(file_path, 'w') as file:\n",
+    "        for item in data:\n",
+    "            file.write(json.dumps(item) + '\\n')\n",
+    "\n",
+    "# Define splits\n",
+    "splits = [\"train\", \"validation\", \"test\"]\n",
+    "\n",
+    "# Save splits to JSONL files and calculate their sizes\n",
+    "for split in splits:\n",
+    "    if split in dataset:\n",
+    "        save_to_jsonl(file_paths[split], dataset[split])\n",
+    "    else:\n",
+    "        print(f\"Split {split} not found in the dataset.\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2d0cc359-0598-40aa-af80-9503ecd4dac1",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "---\n",
+    "##  Step-by-step instructions\n",
+    "\n",
+    "This workflow is structured into seven notebooks:\n",
+    "1. [Prepare the dataset](./01_data_preparation.ipynb)\n",
+    "2. [Finetune the teacher on the dataset](./02_teacher_finetuning.ipynb)\n",
```

Comment:
fix punctuation

 "2. [Fine-tune the teacher on the dataset](./02_teacher_finetuning.ipynb).\n",